Fakeifying views shouldnt create symbols when dynamic=False

bdhirsh opened the pull request:
Fixes https://github.com/pytorch/pytorch/issues/123298

I was also seeing some crashes in torchtrain due to dynamic shapes, even when I set `compile(dynamic=False)` (cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @wanchaol). This doesn't fix the underlying dynamic shape issues with compile + DTensor, but it does prevent dynamic shapes from leaking in.

cc @jbschlosser for review

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #123347
* __->__ #123348
* #122751
* #122502

bdhirsh committed:
Fakeifying views shouldnt create symbols when dynamic=False

[ghstack-poisoned]

bdhirsh committed:
Update on "Fakeifying views shouldnt create symbols when dynamic=False"




[ghstack-poisoned]

jbschlosser commented:
Left a comment on the issue: https://github.com/pytorch/pytorch/issues/123298#issuecomment-2037864731

I'm curious if it's possible to address this by just fixing up the logic for simplifying out symbols

ezyang reviewed (approved):
This is fine but our "ShapeEnv exists but it is still static" handling is still mildly cursed. See https://github.com/pytorch/pytorch/pull/121855 for a refactor that I thought would work but didn't.

bdhirsh committed:
Update on "Fakeifying views shouldnt create symbols when dynamic=False"

Fixes https://github.com/pytorch/pytorch/issues/123298

I was also seeing some crashes in torchtrain due to dynamic shapes, even when I set `compile(dynamic=False)` (cc wanchaol). This doesn't fix the underlying dynamic shape issues with compile + DTensor, but it does prevent dynamic shapes from leaking in.

cc jbschlosser for review




[ghstack-poisoned]

bdhirsh committed:
Update on "Fakeifying views shouldnt create symbols when dynamic=False"

Fixes https://github.com/pytorch/pytorch/issues/123298

I was also seeing some crashes in torchtrain due to dynamic shapes, even when I set `compile(dynamic=False)` (cc wanchaol). This doesn't fix the underlying dynamic shape issues with compile + DTensor, but it does prevent dynamic shapes from leaking in.

cc jbschlosser for review




[ghstack-poisoned]

bdhirsh commented:
@pytorchbot merge
